src/openms/include/OpenMS/MATH/STATISTICS/BasicStatistics.h

```diff
@@ -132,6 +132,12 @@ namespace OpenMS
           variance_ += *iter * diff;
         }
         variance_ /= sum_;
+
+        if (sum_ == 0 && (std::isnan(mean_) || std::isinf(mean_)) )
```

Comment:
there are two places where this could be added in the header

Reply:
why is the sum_ == 0 required?
would std::isnan(mean_) || std::isinf(mean_) be enough?

Reply:
I wanted to avoid fixing stuff that shouldn't be fixed. If those numbers are `nan` because there is a `nan` in the given input, then returning a `0` would be wrong. We should make sure we only set to `0` when it's obvious the `nan` was caused by the calculations in this class.